pkg/manager/member/tidb_member_manager.go

```diff
@@ -93,7 +93,11 @@ func (tmm *tidbMemberManager) Sync(tc *v1alpha1.TidbCluster) error {
 	}
 
 	// Sync Tidb StatefulSet
-	return tmm.syncTiDBStatefulSetForTidbCluster(tc)
+	if err := tmm.syncTiDBStatefulSetForTidbCluster(tc); err != nil {
+		return err
```

Comment:
I'm a bit worried that if something goes wrong with sts synchronization, then the service will not be synchronized too.

Reply:
Yes, such dangers are everywhere in today's tidb-controller-manager because of synchronized reconciling, this is a systematic improvement that I'd like to collect efforts in a dedicated ticket

Reply:
follow up: https://github.com/pingcap/tidb-operator/issues/1245